nativescript-angular/router/ns-location-strategy.ts

```diff
@@ -92,70 +121,76 @@ export class NSLocationStrategy extends LocationStrategy {
 
     pushStateInternal(state: any, title: string, url: string, queryParams: string): void {
         const urlSerializer = new DefaultUrlSerializer();
-        const stateUrlTree: UrlTree = urlSerializer.parse(url);
-        const rootOutlets = stateUrlTree.root.children;
-
-        this.currentUrlTree = stateUrlTree;
-
-        // Handle case where the user declares a component at path "/".
-        // The url serializer doesn't parse this url as having a primary outlet.
-        if (!Object.keys(rootOutlets).length) {
-            const outletStates = this.statesByOutlet["primary"] = this.statesByOutlet["primary"] || [];
-            const isNewPage = outletStates.length === 0;
-            outletStates.push({
-                state: state,
-                title: title,
-                queryParams: queryParams,
-                segmentGroup: stateUrlTree.root,
-                isRootSegmentGroup: true,
-                isPageNavigation: isNewPage,
-                isModalNavigation: false
-            });
-            this.currentOutlet = "primary";
-        }
-
-        Object.keys(rootOutlets).forEach(outletName => {
-            const outletStates = this.statesByOutlet[outletName] = this.statesByOutlet[outletName] || [];
-            const isNewPage = outletStates.length === 0;
-            const lastState = this.peekState(outletName);
-
-            if (!lastState || lastState.segmentGroup.toString() !== rootOutlets[outletName].toString()) {
-                outletStates.push({
-                    state: state,
-                    title: title,
-                    queryParams: queryParams,
-                    segmentGroup: rootOutlets[outletName],
-                    isRootSegmentGroup: false,
-                    isPageNavigation: isNewPage,
-                    isModalNavigation: false
+        this.currentUrlTree = urlSerializer.parse(url);
+        const urlTreeRoot = this.currentUrlTree.root;
+
+        if (!Object.keys(urlTreeRoot.children).length) {
```

Comment:
To reduce indentation of the main case instead of:
``` ts
if (!Object.keys(urlTreeRoot.children).length) {

} else {
    // else clause statements go here...
}
```
Use:
``` ts
if (!Object.keys(urlTreeRoot.children).length) {
    // ...
    return;
}

// else clause statements go here...